server.js

```diff
@@ -2,31 +2,182 @@ import express from 'express'
+    ))
+  }
+  // PAGINATION (Query by page and/or page-size):
+  // -Page (minValue 0, default 0): number to indicate what page to return.
+  // -PageSize (default: 20)
+  const totalNumberOfBooks = booksList.length
+  const firstPageIndex = 0
+  const startIndex = page * pageSize
+  const endIndex = startIndex + +pageSize // converts pageSize (string) to a number.
+  const bookPage = booksList.slice(startIndex, endIndex) //Creating pages using slice.
+  const returnObject = {
+    totalNumberOfBooks: totalNumberOfBooks,
+    pageSize: pageSize,
+    page: page,
+    firstPageIndex: firstPageIndex,
+    lastPageIndex: parseInt(booksList.length / pageSize),
+    results: bookPage
+  }
+
+  if (bookPage.length === 0) {
+    res.status(404).json(ERROR_MESSAGE_DATA_NOT_FOUND)
+  } else if (!page || !pageSize) {
+    res.json(booksList)
+  } else {
+    res.json(returnObject)
+  }
+})
+
+// LANGUAGE-ENDPOINT. // According to Maks friday lecture: language should rather be query param??
+// Get books in a specified language (language_code):
+// Existing language-codes: 'eng', 'en-US', 'spa', 'fre', 'en-GB', 'mul', 'ger', 'ara', 'por', 'grc'
+app.get('/books/language/:language', (req, res) => {
+  const language = req.params.language
+  const booksByLanguage = booksData.filter((book) => book.language_code === language)
+  
+  // I used existingLanguages to console.log which unique languages that exists in the data.
+  // Would be nice to show to the user somehow which languages could be shown, but I didn't come that far.
+  // const existingLanguages = [...new Set(booksData.map(x => x.language_code))] 
+
+  if (booksByLanguage.length > 0) {
+    res.send(booksByLanguage)
+  } else {
+    res.status(404).json(ERROR_MESSAGE_LANGUAGE_NOT_FOUND)
+  }
+})
+
+// ID-ENDPOINT. 
+// Get a single book by id:
+app.get('/books/id/:id', (req, res) => {
+  const id = req.params.id
+  const singleBookId = booksData.find((book) => book.bookID === +id) // + turns string to number
+
+  if (!singleBookId) {
+    res.status(404).json(ERROR_MESSAGE_BOOK_ID_NOT_FOUND)
+  } else {
+    res.json(singleBookId)
+  }
+})
+
+//ISBN-ENDPOINT.
+//Return single book by isbn/isbn13:
+app.get('/books/isbn/:isbn', (req, res) => {
+  const isbn = req.params.isbn
+  const singleBookIsbn = booksData.find((book) => (
+    book.isbn === +isbn || book.isbn13 === +isbn
+  ))
+
+  if (!singleBookIsbn) {
+    res.status(404).json(ERROR_MESSAGE_BOOK_ISBN_NOT_FOUND)
+  } else {
+    res.json(singleBookIsbn)
+  }
+})
+
+// TOP-RATED-BOOKS-ENDPOINT.
+// Get a list of the top 100 rated books:
+// http://localhost:8080/books/ratings/top100
+app.get('/books/ratings/:top100', (req, res) => {
+  const sortedBooks = booksData.sort((a, b) => b.average_rating - a.average_rating)
+  const top100Array = sortedBooks.slice(0, 100)
+
+  if (top100Array.length === 0) {
+    res.status(404).json(ERROR_MESSAGE_RATING_NOT_FOUND)
+  } else {
+    res.json(top100Array)
+  }
+})
```

Comment:
This could also be a query under your main endpoint, something like:
`/books?sort=descending`